modules/oe_multilingual_url_suffix/src/Plugin/LanguageNegotiation/LanguageNegotiationUrlSuffix.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationUrl;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class LanguageNegotiationUrlSuffix for identifying language via URL suffix.
+ *
+ * @LanguageNegotiation(
+ *   id = \Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation\LanguageNegotiationUrlSuffix::METHOD_ID,
+ *   types = {\Drupal\Core\Language\LanguageInterface::TYPE_INTERFACE,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_CONTENT,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_URL},
+ *   weight = -10,
+ *   name = @Translation("URL suffix"),
+ *   description = @Translation("Language from the URL (Path suffix)."),
+ *   config_route_name = "oe_multilingual_url_suffix.negotiation_url_suffix"
+ * )
+ */
+class LanguageNegotiationUrlSuffix extends LanguageNegotiationUrl {
+
+  /**
+   * The language negotiation method id.
+   */
+  const METHOD_ID = 'language-url-suffix';
```

Comment:
I'd prefix this with the module name too.

Reply:
I have added prefix with the module name but for avoiding repeating inside name reamed to `oe-multilingual-url-suffix-negotiation-method`.
Let me know if this is ok for you.

Reply:
Let's use a simpler `oe-multilingual-url-suffix` since it's clear it's a language negotiation method.